src/ert/cli/model_factory.py

```diff
@@ -129,7 +135,37 @@ def _setup_ensemble_experiment(
             minimum_required_realizations=config.analysis_config.minimum_required_realizations,
             ensemble_size=config.model_config.num_realizations,
             stop_long_running=config.analysis_config.stop_long_running,
-            experiment_name=args.experiment_name,
+            experiment_name=experiment_name,
+        ),
+        config,
+        storage,
+        config.queue_config,
+    )
+
+
+def _setup_evaluate_parameters(
+    config: ErtConfig, storage: StorageAccessor, args: Namespace
+) -> EvaluateEnsemble:
+    min_realizations_count = config.analysis_config.minimum_required_realizations
+    active_realizations = _realizations(args, config.model_config.num_realizations)
+    active_realizations_count = int(np.sum(active_realizations))
+    if active_realizations_count < min_realizations_count:
+        config.analysis_config.minimum_required_realizations = active_realizations_count
+        ConfigWarning.ert_context_warn(
+            f"Due to active_realizations {active_realizations_count} is lower than "
+            f"MIN_REALIZATIONS {min_realizations_count}, MIN_REALIZATIONS has been "
+            f"set to match active_realizations.",
+        )
+
+    return EvaluateEnsemble(
+        EvaluateParametersRunArguments(
+            random_seed=config.random_seed,
+            active_realizations=active_realizations.tolist(),
+            current_case=args.ensemble_name,
+            minimum_required_realizations=config.analysis_config.minimum_required_realizations,
+            ensemble_size=config.model_config.num_realizations,
+            stop_long_running=config.analysis_config.stop_long_running,
+            experiment_name=None,
```

Comment:
What does a `None` experiment name mean?

Reply:
We never create an experiment here, so it does not make sense, but I moved it to __post_init__ instead 👍

Reply:
Remember now, is because it is required on the class we inherit from, so need to provide it to the constructor.